test/services/messengerSync/utils.test.js

```diff
@@ -0,0 +1,121 @@
+		is_school_admin: true,
+		is_school_teacher: true,
+    },
+    room: {
+		id: 1234566,
+		name: 'Mathe 6b',
+		type: 'course',
+		is_moderator: false,
+		bidirectional: true
+    }
+} */
+
+	describe('buildAddUserMessage', () => {
+		it('builds a correct object for teacher with course', async () => {
+			this.app = app;
+			const school = await testObjects.createTestSchool({ features: ['messenger'] });
+			const teacher = await testObjects.createTestUser({ roles: ['teacher'], schoolId: school._id });
+			const course = await testObjects.createTestCourse({ teacherIds: [teacher._id], schoolId: school._id });
+			const result = await buildAddUserMessage({ userId: teacher._id, courses: [course] });
+			expect(result.method).to.equal('adduser');
+
+			expect(result.school.id).to.equal(school._id.toString());
+			expect(result.school.has_allhands_channel).to.equal(true);
+			expect(result.school).to.haveOwnProperty('name');
+
+			expect(result.user).to.haveOwnProperty('name');
+			expect(result.user).to.haveOwnProperty('id');
+			expect(result.user.is_school_admin).to.equal(false);
+			expect(result.user.is_school_teacher).to.equal(true);
+
+			expect(Array.isArray(result.rooms)).to.equal(true);
+			expect(result.rooms.length).to.eq(1);
+			const room = result.rooms[0];
+			expect(room.id).to.equal(course._id.toString());
+			expect(room.type).to.equal('course');
+			expect(room.is_moderator).to.equal(true);
+			expect(room).to.haveOwnProperty('bidirectional');
+			expect(room).to.haveOwnProperty('name');
+		});
+
+		it('builds a correct object for teacher with team', async () => {
+			this.app = app;
+			const school = await testObjects.createTestSchool({ features: ['messenger'] });
+			const { user, team } = await testObjects.createTestTeamWithOwner(
+				{ roles: ['teacher'], schoolId: school._id },
+			);
+			const result = await buildAddUserMessage({ userId: user._id, teams: [team] });
+			expect(result.method).to.equal('adduser');
+
+			expect(result.school.id).to.equal(school._id.toString());
+			expect(result.school.has_allhands_channel).to.equal(true);
+			expect(result.school).to.haveOwnProperty('name');
+
+			expect(result.user).to.haveOwnProperty('name');
+			expect(result.user).to.haveOwnProperty('id');
+			expect(result.user.is_school_admin).to.equal(false);
+			expect(result.user.is_school_teacher).to.equal(true);
+
+			expect(Array.isArray(result.rooms)).to.equal(true);
+			expect(result.rooms.length).to.eq(1);
+			const room = result.rooms[0];
+			expect(room.id).to.equal(team._id.toString());
+			expect(room.type).to.equal('team');
+			expect(room.is_moderator).to.equal(true);
+			expect(room).to.haveOwnProperty('bidirectional');
+			expect(room).to.haveOwnProperty('name');
+		});
+
+		it('builds a correct object for schoolSync', async () => {
+			this.app = app;
+			const school = await testObjects.createTestSchool({ features: ['messenger'] });
+			const { user, team } = await testObjects.createTestTeamWithOwner(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['team' is assigned a value but never used.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=5203703)